portal/static/js/main.js

```diff
@@ -5166,16 +5166,36 @@ var tnthAjax = {
             };
         };
         if (!params) params = {};
-        this.sendRequest('/api/user/'+userId+'/notification/'+notificationId, 'DELETE', userId, {"sync":params.sync}, function(data) {
-            if (data) {
-                if (!data.error) {
-                    if (callback) callback(data);
-                } else {
-                    if (callback) callback({"error": i18next.t("Error occurred deleting notification.")});
+
+        var self = this;
+
+        this.getNotification(userId, false, function(data) {
+            if (data.notifications) {
+                 /* 
+                 * check if there is notification for this id
+                 * dealing with use case where user deletes same notification in a separate open window
+                 */
+                var arrNotification = $.grep(data.notifications, function(notification) {
+                    return notification.id == notificationId;
+                });
+                if (arrNotification.length > 0) {
+                    /* 
+                     * delete notification only if it exists
+                     */
+                    self.sendRequest('/api/user/'+userId+'/notification/'+notificationId, 'DELETE', userId, {"sync":params.sync}, function(data) {
+                        if (data) {
```

Comment:
Would it make sense to check `if (!callback) return;` once, at the beginning of this function? Then you wouldn't need to each time?

Reply:
@ivan-c Ivan, ah, you are right - I see multiple checks for the existence of callback - makes sense to just check for it once.  I did a slight modification of your suggestion - assigning callback to an anonymous function if it is not provided, in case the callee didn't provide it but still wants the function executed without callback.

Reply:
Thanks for giving it a try! 
I've seen in the cPRO codebase that a pile of if/else only gets bigger over time. I try to avoid [deeply nesting if/else conditions](https://refactoring.com/catalog/replaceNestedConditionalWithGuardClauses.html) when I can, but I wasn't sure if it made sense here.